hetio/neo4j.py

```diff
@@ -434,20 +438,16 @@ def construct_pdp_query(metarels, dwpc=None, property='name', join_hint='midpoin
             [
             {degree_query}
             ] AS degrees, path
-            WITH sum(reduce(pdp = 1.0, d in degrees| pdp * d ^ -{{ w }})) as DWPC
-
-            MATCH path = {metapath_query}{using_query}
-            WHERE n0.{property} = {{ source }}
-            AND n{length}.{property} = {{ target }}{unique_nodes_query}
-            WITH
-            [
-            {degree_query}
-            ] AS degrees, path, DWPC
-            WITH path, DWPC, reduce(pdp = 1.0, d in degrees| pdp * d ^ -{{ w }}) as PDP
+            WITH path, reduce(pdp = 1.0, d in degrees| pdp * d ^ -{{ w }}) AS PDP
+            WITH path, collect(PDP) AS pdps, PDP
+            WITH collect({{path: path, pdps: pdps}}) AS allData, sum(PDP) AS DWPC
```

Comment:
I think we can improve on the variable names from those from the [stackoverflow answer](https://stackoverflow.com/a/54245650/4651668)